### PR TITLE
MNT: Removing 3.4 deprecations

### PR DESCRIPTION
**Pull request by greglucas**
## PR Summary

This removes more 3.4 deprecations. I've put each removal into a separate commit, this seemed easier than opening a bunch of PRs, but I can do that if it would be easier to review.

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: doc/api/next_api_changes/removals/23XXX-GL.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Get/set window title methods have been removed from the canvas
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the corresponding methods on the FigureManager instead.
```
Comment (timhoffm) [inline doc/api/next_api_changes/removals/23XXX-GL.rst]: Maybe mention also that one can use the native methods of a GUI framework if cross-backend compatibility is not needed.

File: doc/api/next_api_changes/removals/23XXX-GL.rst [hunk 2]
```diff
@@ -15,3 +15,8 @@ The **minimum_descent** has been removed from ``TextArea``
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The minimum_descent is now effectively always True.
+
+Axes creation no longer accepts positional arguments other than figure and rect
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pass in the explicit keyword arguments desired when creating the axes.
```
Comment (timhoffm) [inline doc/api/next_api_changes/removals/23XXX-GL.rst]: Wording: explicit keyword arguments does not speak to me. Better: "Pass all other parameters as keyword arguments."

File: doc/api/next_api_changes/removals/23XXX-GL.rst [hunk 3]
```diff
@@ -25,3 +25,8 @@ Pass in the explicit keyword arguments desired when creating the axes.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Use ``docutils.parsers.rst.directives.images.Image.align`` instead.
+
+``imread()`` no longer accepts URLs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These should be opened and passed directly to Pillow now.
```
Comment (timhoffm) [inline doc/api/next_api_changes/removals/23XXX-GL.rst]: Please take over the example.

**Review by timhoffm — APPROVED** [review]

File: doc/api/next_api_changes/removals/23XXX-GL.rst (hunk 3 above)
Comment (greglucas) [reply doc/api/next_api_changes/removals/23XXX-GL.rst]: :+1: I updated all of the change log suggestions. I went back and took the wording from the 3.4 deprecations (not sure why I didn't do that in the first place :smile:)

**Review by jklymak — APPROVED** [review]

**Pull request merged**